asgs_main.sh

```diff
@@ -75,34 +75,65 @@ checkFileExistence()
   if [[ -z $FNAME ]]; then
      fatal "$THIS: The $FTYPE was not specified in the configuration file. When it is specified, the ASGS will look for it in the path ${FPATH}."
   fi
-  success=no
+  local success=no
   if [ $FNAME ]; then
      if [ -e "${FPATH}/${FNAME}" ]; then
         logMessage "$THIS: The $FTYPE '${FPATH}/${FNAME}' was found."
         success=yes
      else
-        # if this is a mesh (fort.14), nodal attributes (fort.13), static water level correction,
-        # or self attracting / earth load tide (fort.24) file, attempt to download and uncompress it
-        if [[ $FTYPE == "ADCIRC mesh file"  ]]; then
-           logMessage "Downloading $FTYPE from ${MESHURL}/${FNAME}.xz."
-           curl --version >> $SYSLOG
-           curl ${MESHURL}/${FNAME}.xz > ${FPATH}/${FNAME}.xz 2> errmsg || warn "$THIS: Failed to download mesh from ${MESHURL}/${FNAME}.xz to ${FPATH}/${FNAME}.xz: `cat errmsg`."
+        # If this is a mesh (fort.14), nodal attributes (fort.13), static water level correction,
+        # or self attracting / earth load tide (fort.24) file, attempt to download and uncompress it.
+        # If the URL starts with "scp://" then the Operator's ssh configuration must support
+        # public key authentication with the host that the files will be downloaded from.
+        # The URLs used here have their default values set in config/mesh_defaults.sh and these
+        # URL values can be overridden in the Operator's ~/.asgsh_profile file or in the
+        # configuration file for the ASGS instance.
+        # If scp is to be used to download files, this function expects URLs to be in the form
+        #    scp://tacc_tds3/meshes
+        # where the directory ("meshes" in the above case) is assumed by scp to be relative
+        # to the Operator's home directory on the remote machine unless the path starts with
+        # a forward slash:
+        #    scp://tacc_tds3//meshes
+        # In which case it is treated as a full path.
+        local URL
+        case $FTYPE in
+           "ADCIRC mesh file")
+              URL=$MESHURL
+              ;;
+           "ADCIRC nodal attributes (fort.13) file")
+              URL=$NODALATTRIBUTESURL
+              ;;
+           "ADCIRC static water level offset data file")
+              URL=$OFFSETURL
+              ;;
+           "ADCIRC self attracting earth load tide file")
+              URL=$LOADTIDEURL
+              ;;
+           *)
+              warn "$THIS: Unrecognized file type to download: '$FTYPE'."
+              URL="unknown"
+              ;;
+        esac
+        local downloadCMD
+        if [[ $URL =~ "http://" || $URL =~ "https://" ]]; then
+           logMessage "$THIS: The curl version is $(curl --version)"
+           downloadCMD="curl ${URL}/${FNAME}.xz --output ${FPATH}/${FNAME}.xz"
+        elif [[ $URL =~ "scp://" ]]; then
```

Comment:
Sent you a DM about this one, `scp://` may not be what we want. And if it is this is not looking sufficient to handle it like how you mean here (via `scp`).

Reply:
See my latest message, `scp://` seems legit.